include/SABER/SaberCondAllocator.h

```diff
@@ -144,10 +144,54 @@ class SaberCondAllocator
         return negConds.test(id);
     }
 
-    /// Get dominators
-    inline DominatorTree* getDT(const Function* fun)
+    inline const bool pdtDominates(const BasicBlock* bbKey, const BasicBlock* bbValue)
     {
-        return cfInfoBuilder.getDT(fun);
+        return dominates(bbKey,bbValue,SVFFunction::DtK::PostDominatorTree);
+    }
+
+    inline const bool dtDominates(const BasicBlock* bbKey, const BasicBlock* bbValue)
+    {
+        return dominates(bbKey,bbValue,SVFFunction::DtK::DominatorTree);;
+    }
+
+    inline const bool dominates(const BasicBlock* bbKey, const BasicBlock* bbValue, const SVFFunction::DtK dtType)
+    {
+        if (bbKey == bbValue)
+            return true;
+        
+        const SVFFunction*  keyFunc = LLVMModuleSet::getLLVMModuleSet()->getSVFFunction(bbKey->getParent());
+        const SVFFunction*  valueFunc = LLVMModuleSet::getLLVMModuleSet()->getSVFFunction(bbValue->getParent());
+        // An unreachable node is dominated by anything.
```

Comment:
put an assertion "assert((keyFunc == valueFunc) && "two basicblocks should be in the same function!")"